src/inmanta/deploy/scheduler.py

```diff
@@ -345,18 +400,43 @@ async def _new_version(
 
     def _start_for_agent(self, agent: str) -> None:
         """Start processing for the given agent"""
-        self._workers[agent] = asyncio.create_task(self._run_for_agent(agent))
+        self._workers[agent] = TaskRunner(endpoint=agent, scheduler=self)
+        self._workers[agent].start()
 
-    async def _run_for_agent(self, agent: str) -> None:
-        """Main loop for one agent"""
-        while self._running:
-            task: Task = await self._work.agent_queues.queue_get(agent)
-            try:
-                await task.execute(self, agent)
-            except Exception:
-                LOGGER.exception("Task %s for agent %s has failed and the exception was not properly handled", task, agent)
+    async def should_be_running(self, endpoint: str) -> bool:
+        """
+        Return if the agent (or the Scheduler if endpoint == Scheduler id) should be running. This will also
+            check if the environment is halted
```

Comment:
I like the "authoritative entity" phrasing from the other docstring. What about:
```suggestion
        Check in the DB (authoritative entity) if the agent (or the Scheduler if endpoint == Scheduler id) should be running i.e. if it is not paused and its environment is not halted.
```